prime-router/src/main/kotlin/serializers/Hl7Serializer.kt

```diff
@@ -924,6 +924,16 @@ class Hl7Serializer(val metadata: Metadata) : Logging {
         terser.set("/PATIENT_RESULT/ORDER_OBSERVATION/OBSERVATION/OBX-23-7", "XX")
     }
 
+    private fun setTruncationLimitWithEncoding(value: String, truncationLimit: Int): Int {
```

Comment:
Need kdoc

Reply:
Add unit tests for this method.  What happens with 0, 1 and many escape characters (not sure if many are allowed).